packages/core/src/util/esm.js

```diff
@@ -66,3 +66,31 @@ exports.isESMApp = function isESMApp() {
 
   return isESM;
 };
+
+exports.tracerInstrumentationInfo = function tracerInstrumentationInfo() {
```

Comment:
This is only applicable for the collector package? what about the serverless packages?

Reply:
> Instead of
> 
> includes('@instana/collector')
> 
> can check for
> 
> esm-register.mjs

this won't work btw


> The serverless whole is another scope.

> 

Yeah, I agree.

Reply:
I believe we need to check the flag for all our packages. Checking `@instana/collector` would only apply to the collector package. We already have a similar check in place for detecting ESM, as seen [here](https://github.com/instana/nodejs/blob/main/packages/core/src/util/esm.js#L51). I'm thinking of either extending or referencing this function . This approach would help us avoid maintaining flags in multiple functions. What do you think?

Reply:
That would be a good idea. Initially, I designed the function like this but then later decided to check for a single package for checking serverless and collector specifically.
Re-using the existing function will be a good idea, even I thought of this. But since the outputs and checks differ in the `function isESMApp()` and `function tracerInstrumentationInfo()`, need to decide how can we extend this. I will get back to you after that. Please share if you have any better solutions in mind

Reply:
Implemented a new logic!